plugins/funind/recdef.ml

```diff
@@ -1615,6 +1615,22 @@ let com_eqn uctx nb_arg eq_name functional_ref f_ref terminate_ref
   in
   ()
 
+let check_relation_type env evd relation =
```

Comment:
This is too restrictive, you're asking for a type that is syntactically equal to some relation but you should probably use conversion instead. I'd recommend using `Reductionops.dest_arity`.

Reply:
Indeed, that was rather naive. `Reductionops.dest_arity` keeps let-ins I think, so, we can also adds `whd_all` ourselves. I'll push a new version when CI is done.